api/test/media_integration.py

```diff
@@ -27,14 +27,14 @@ def search_by_category(media_path, category, fixture):
     assert all(audio_item["category"] == category for audio_item in results)
 
 
-def tag_collection(media_path):
-    response = requests.get(f"{API_URL}/v1/{media_path}/tag/cat")
+def tag_collection(media_path, tag="cat"):
+    response = requests.get(f"{API_URL}/v1/{media_path}/tag/{tag}")
     assert response.status_code == 200
 
     results = response.json()["results"]
     for r in results:
         tag_names = [tag["name"] for tag in r["tags"]]
-        assert "cat" in tag_names
+        assert tag in tag_names
```

Comment:
I guess it isn't sensible to do `assert tag.upper() in tag_names` because both `Cat` and `cat` could be in the same list of tags :thinking: Just trying to think of how to really make the tests check "this matched only `cat` and not any other capitalisation variant". Maybe combining the tests and doing `[tag, tag.lower(), tag.upper()]` as individual queries and then asserting each one is a different result set (and that the tag as searched casing is in the list of tags)?

Reply:
When you consider that an item can have both a lower and an upper case tag, adding more checks seems to be overcomplicating things. We do run two tests for lower and upper case tags, and confirm that the matches are of the same casing, so I think it's enough?